### ISSUE TITLE
calling subprocess within python to run succession of commands makes predict file one row too small

### ISSUE THREAD

**Issue by dhruvghulati** [JohnLangford/vowpal_wabbit#1068]
Command line prediction is predicting fine on my files, but when I run commands from within python like so:

```
trainCommand = vw --csoaa 16 data/output/cost/closed_cost_1.dat -f data/output/cost/csoaa.model
predictCommand = vw -t -i data/output/cost/csoaa.model data/output/cost/test.dat -p data/output/cost/closed_cost_1.predict

subprocess.call(trainCommand.split(' '))
subprocess.call(predictCommand.split(' '))
```

I get a predict file that only goes to 498 rows, instead of 499 as it does on the command line.

[closed_cost_1.predict.txt](https://github.com/JohnLangford/vowpal_wabbit/files/411507/closed_cost_1.predict.txt)
[test.dat.txt](https://github.com/JohnLangford/vowpal_wabbit/files/411508/test.dat.txt)
[closed_cost_1.dat.txt](https://github.com/JohnLangford/vowpal_wabbit/files/411743/closed_cost_1.dat.txt)

Note, the above lines of trainCommand and predictCommand are generated in a loop i.e. I have many files numbered differently with different cost combinations e.g. 

```
trainCommand = vw --csoaa 16 data/output/cost/closed_cost_2.dat -f data/output/cost/csoaa.model
predictCommand = vw -t -i data/output/cost/csoaa.model data/output/cost/test.dat -p data/output/cost/closed_cost_2.predict
```

Running exact files on command line:

[closed_cost_1.predict.txt](https://github.com/JohnLangford/vowpal_wabbit/files/411518/closed_cost_1.predict.txt)

**Comment by JohnLangford** [comment]
No idea what's happening here.  @hal3 @gramhagen any ideas?

**Comment by gramhagen** [comment]
I can't reproduce the problem with the latest code.  You mention you're looping through various files, could it be a problem there?

`Scotts-MBP:vw_test sgraham$ ls
closed_cost_1.dat.txt	test.dat.txt
Scotts-MBP:vw_test sgraham$ vw --csoaa 16 closed_cost_1.dat.txt -f csoaa.model
final_regressor = csoaa.model
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = closed_cost_1.dat.txt
num sources = 1
average    since         example     example  current  current  current
loss       last          counter      weight    label  predict features
0.115386   0.115386          1      1.0    known        1        3
0.057693   0.000000          2      2.0    known        6        3
0.028846   0.000000          4      4.0    known        6        3
0.249717   0.470588          8      8.0    known        5       14
0.471171   0.692625         16     16.0    known       15        6
0.327593   0.184014         32     32.0    known       10       19
0.342268   0.356942         64     64.0    known       11       24
0.391684   0.441100        128    128.0    known        4       19
0.427318   0.462952        256    256.0    known        4        6
0.408848   0.390378        512    512.0    known        1        5
0.565312   0.721776       1024   1024.0    known       15       15
0.572700   0.580088       2048   2048.0    known        1        6
0.508197   0.443693       4096   4096.0    known        4       16
0.600807   0.693417       8192   8192.0    known       15       35

finished run
number of examples per pass = 15000
passes used = 1
weighted example sum = 15000
weighted label sum = 0
average loss = 0.632362
total feature number = 353228
Scotts-MBP:vw_test sgraham$ vw -t -i csoaa.model test.dat.txt -p closed_cost_1.predict
only testing
predictions = closed_cost_1.predict
Num weight bits = 18
learning rate = 10
initial_t = 1
power_t = 0.5
using no cache
Reading datafile = test.dat.txt
num sources = 1
average    since         example     example  current  current  current
loss       last          counter      weight    label  predict features
0.000000   0.000000          1      1.0    unknown        4       13
0.000000   0.000000          2      2.0    unknown        4       14
0.000000   0.000000          4      4.0    unknown        4       11
0.000000   0.000000          8      8.0    unknown        4       11
0.000000   0.000000         16     16.0    unknown       13       28
0.000000   0.000000         32     32.0    unknown        5       31
0.000000   0.000000         64     64.0    unknown        4        5
0.000000   0.000000        128    128.0    unknown        4       13
0.000000   0.000000        256    256.0    unknown        4       18

finished run
number of examples per pass = 500
passes used = 1
weighted example sum = 500
weighted label sum = 0
average loss = 0
total feature number = 8004
Scotts-MBP:vw_test sgraham$ wc -l closed_cost_1.predict
     500 closed_cost_1.predict`

`In [1]: import subprocess

In [2]: trainCommand = 'vw --csoaa 16 closed_cost_1.dat.txt -f csoaa.model.py'
   ...: predictCommand = 'vw -t -i csoaa.model.py test.dat.txt -p closed_cost_1.predict.py'
   ...:
   ...: subprocess.call(trainCommand.split(' '))
   ...: subprocess.call(predictCommand.split(' '))
   ...:
final_regressor = csoaa.model.py
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = closed_cost_1.dat.txt
num sources = 1
average    since         example     example  current  current  current
loss       last          counter      weight    label  predict features
0.115386   0.115386          1      1.0    known        1        3
0.057693   0.000000          2      2.0    known        6        3
0.028846   0.000000          4      4.0    known        6        3
0.249717   0.470588          8      8.0    known        5       14
0.471171   0.692625         16     16.0    known       15        6
0.327593   0.184014         32     32.0    known       10       19
0.342268   0.356942         64     64.0    known       11       24
0.391684   0.441100        128    128.0    known        4       19
0.427318   0.462952        256    256.0    known        4        6
0.408848   0.390378        512    512.0    known        1        5
0.565312   0.721776       1024   1024.0    known       15       15
0.572700   0.580088       2048   2048.0    known        1        6
0.508197   0.443693       4096   4096.0    known        4       16
0.600807   0.693417       8192   8192.0    known       15       35

finished run
number of examples per pass = 15000
passes used = 1
weighted example sum = 15000
weighted label sum = 0
average loss = 0.632362
total feature number = 353228
only testing
predictions = closed_cost_1.predict.py
Num weight bits = 18
learning rate = 10
initial_t = 1
power_t = 0.5
using no cache
Reading datafile = test.dat.txt
num sources = 1
average    since         example     example  current  current  current
loss       last          counter      weight    label  predict features
0.000000   0.000000          1      1.0    unknown        4       13
0.000000   0.000000          2      2.0    unknown        4       14
0.000000   0.000000          4      4.0    unknown        4       11
0.000000   0.000000          8      8.0    unknown        4       11
0.000000   0.000000         16     16.0    unknown       13       28
0.000000   0.000000         32     32.0    unknown        5       31
0.000000   0.000000         64     64.0    unknown        4        5
0.000000   0.000000        128    128.0    unknown        4       13
0.000000   0.000000        256    256.0    unknown        4       18

finished run
number of examples per pass = 500
passes used = 1
weighted example sum = 500
weighted label sum = 0
average loss = 0
total feature number = 8004
Out[2]: 0

In [3]: !wc -l closed_cost_1.predict.py
     500 closed_cost_1.predict.py`

**Comment by JohnLangford** [comment]
Closing this---issue likely solved.
